src/main/java/com/jakewharton/retrofit2/adapter/kotlin/coroutines/experimental/CoroutineCallAdapterFactory.kt

```diff
@@ -110,6 +112,37 @@ class CoroutineCallAdapterFactory private constructor() : CallAdapter.Factory()
     }
   }
 
+  private object EmptyBodyCallAdapter : CallAdapter<Any, Deferred<Unit>> {
+
+    override fun responseType() = Unit::class.java
+
+    override fun adapt(call: Call<Any>): Deferred<Unit> {
+      val deferred = CompletableDeferred<Unit>()
+
+      deferred.invokeOnCompletion {
```

Comment:
I'm not sure this works in recent versions of kotlinx.coroutines. I wrote a test to try to verify this kind of behavior, and I couldn't get it to pass. After digging further, this calls its internal counterpart with [`onCancelling = false`](https://github.com/Kotlin/kotlinx.coroutines/blob/master/common/kotlinx-coroutines-core-common/src/JobSupport.kt#L402), so I don't think this handler will ever be called. The only way I was able to get something _like_ this to work (without calling coroutines internal API) was to do:
```
try {
    val result = deferred.await()
} catch (e: CancellationException {
    cancellable.cancel()
    throw e
}
```
at the point of use of the `Deferred` object.